### PR TITLE
model: add hidden to ColumnInfo

### PR DESCRIPTION
**Pull request by wjhuang2016**
### What problem does this PR solve? 
To support `hidden column`, we need to add a field in ColumnInfo to indicate if the column is hiddened. A hidden column can't be selected, inserted, updated or deleted.
Besies, `SHOW EXTENDED COLUMNS FROM XXX` can read the hidden column, support it here.

### What is changed and how it works?
Add a filed in ColumnInfo, update parser to support `EXTENDED` in show columns statement.

### Check List 

Tests 

 - N/A

Code changes

 - Has exported variable/fields change

### CODE REVIEW

File: parser_test.go [hunk 1]
```diff
@@ -866,6 +866,10 @@ func (s *testParserSuite) TestDBAStmt(c *C) {
 		// for show full columns
 		{"show columns in t;", true, "SHOW COLUMNS IN `t`"},
 		{"show full columns in t;", true, "SHOW FULL COLUMNS IN `t`"},
+		// for show extended columns
+		{`SHOW COLUMNS FROM City;`, true, "SHOW COLUMNS IN `City`"},
+		{`SHOW EXTENDED COLUMNS FROM City;`, true, "SHOW EXTENDED COLUMNS IN `City`"},
+		{`SHOW EXTENDED FIELDS FROM City;`, true, "SHOW EXTENDED COLUMNS IN `City`"},
```
Comment (djshow832) [inline parser_test.go]: How about `show extended full`?

Comment (wjhuang2016) [reply parser_test.go]: Done

**Comment by djshow832** [comment]
LGTM

File: model/model.go [hunk 2]
```diff
@@ -93,6 +93,7 @@ type ColumnInfo struct {
 	types.FieldType     `json:"type"`
 	State               SchemaState `json:"state"`
 	Comment             string      `json:"comment"`
+	Hidden              bool        `json:"hidden"`
```
Comment (bb7133) [inline model/model.go]: Please add a comment for this attribute.

File: model/model.go [hunk 3]
```diff
@@ -93,6 +93,8 @@ type ColumnInfo struct {
 	types.FieldType     `json:"type"`
 	State               SchemaState `json:"state"`
 	Comment             string      `json:"comment"`
+	// A hidden column is used internally(expression index) and are not accessible by users.
```
Comment (bb7133) [inline model/model.go]:
```suggestion
	// A hidden column is used internally(by expression index) and is not accessible to the users.
```

**Pull request merged**